src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -158,12 +158,6 @@ void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Forward(
   if (parameter.is_empty())
     ResetParameters();
 
-  if (!deterministic)
-  {
-    deterministic = true;
-    ResetDeterministic();
-  }
-
```

Comment:
The forward call shouldn't interfere with deterministic since this will be called either from Train or Predict or directly.

Reply:
@zoq can you see if this is required for any use case? I think we should be able to remove this.

Reply:
That's fine.